portfolio.py

```diff
@@ -2,21 +2,47 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, np_array=None, target=None):
         dict.__init__(self)
-        self.date = date
+        if np_array is not None:
+            if isinstance(np_array, np.ndarray):
+                self.decode_nparray(target, np_array)
+            else:
+                raise TypeError(
+                    'The input is %s, not a numpy array.' % type(np_array))
+
+    def decode_nparray(self, target, np_array):
```

Comment:
This is symmetric to `encode_to_nparray`, so its name should be `decode_from_nparray`.